### ISSUE TITLE
Ionic 5,  Angular 8 - issues with 2.1.3

### ISSUE THREAD

**Issue by hdk** [Cordobo/angularx-qrcode#99]
In a small project with 
ionic 5, angular 8, cordova:

- updating angularx-qrcode to 2.1.3 
- calling: ionic serve
- navigating to page containing the qr-code image
I get this error on entering:
core.js:6014 ERROR Error: Uncaught (in promise): ReferenceError: Cannot access 'CardPageModule' before initialization
ReferenceError: Cannot access 'CardPageModule' before initialization
    at Module.CardPageModule (card-routing.module.ts:17)
    at customer-routing.module.ts:49
    at ZoneDelegate.invoke (zone-evergreen.js:359)
    at Object.onInvoke (core.js:39699)
    at ZoneDelegate.invoke (zone-evergreen.js:358)
    at Zone.run (zone-evergreen.js:124)

customer-routing.module.ts:49:  loadChildren: () =>  import('./card/card.module').then(m => m.CardPageModule)
card-routing.module.ts:17:  export class CardPageRoutingModule {}

- downgrading to angularx-qrcode to 2.1.2 (did not touch anything else)
- calling: ionic serve
everything is ok again 

Any ideas? any help?

BR
H

**Comment by Cordobo** [comment]
Hello @hdk ,

thanks for reporting this issue. Could you do two things:

1) try to install angularx-qrcode in version 2.1.3 again and run ionic serve, just to make sure npm didn't broke the package during installion:

```
npm cache clean
npm  install --save angularx-qrcode@2.1.3
ionic serve
```

2) As only one dependency was changed, please go to `node_modules/angularx-qrcode`and find the line `"qrcode": "1.4.4",`. Now replace it with `"qrcode": "1.4.2",`, save the package.json and run

```
npm install --save qrcode@1.4.2
ionic serve
```

Please let me know if it worked for you or if you still get an error.

**Comment by hdk** [comment]
Hello!

after:
npm cache clean
npm  install --save angularx-qrcode@2.1.3
ionic serve
I still get this crash.

btw: in package.json the new line is
"angularx-qrcode": "^2.1.3",
which will lead to unwanted updates into 2.3.x .... shouldn't that be "~2.1.3" ?

and after:
npm install --save qrcode@1.4.2
ionic serve
.... it works!!  no more crashes.

**Comment by Cordobo** [comment]
Hi @hdk 

regarding your "btw" - yes it would lead to unwanted updates, but as this was only for debugging, it's fine (I guess).

My goal was to find out if qrcode in version 1.4.4 was our culprit. Apparently it is. I will downgrade the dependency and release new versions.

Thanks for your help!

**Comment by hdk** [comment]
Great!

I got this entry in package.json on very first install of version 2.1.3 as well!
"angularx-qrcode": "^2.1.3" 

had to manually change it and downgrade again after a npm update.

**Comment by Cordobo** [comment]
True, one should add the flags `--save --save-exact` if they want a pinned version (or use `npm shrinkwrap`, but that's a different story)

I released version 2.1.4, could you try it if it fixes the initial issue?

**Comment by Cordobo** [comment]
Good to hear.

Regarding the npm, I have a deeper look into it and update the npm strings accordingly.

Closing.
